lib/bolt/cli.rb

```diff
@@ -480,38 +481,30 @@ def outputter
       @outputter ||= Bolt::Outputter.for_format(@config[:format])
     end
 
-    def load_task_data(name, modulepath)
-      module_name, file_name = name.split('::', 2)
-      file_name ||= 'init'
-
+    def load_task_data(name, args, modulepath)
+      task_type = nil
+      task = nil
       begin
-        env = Puppet::Node::Environment.create('bolt', modulepath)
-        Puppet.override(environments: Puppet::Environments::Static.new(env)) do
-          data = Puppet::InfoService::TaskInformationService.task_data(
-            env.name, module_name, name
-          )
-
-          file = data[:files].find { |f| File.basename(f, '.*') == file_name }
-          if file.nil?
-            raise Bolt::CLIError, "Failed to load task file for '#{name}'"
+        Dir.mktmpdir('bolt') do |dir|
+          cli = []
+          Puppet::Settings::REQUIRED_APP_SETTINGS.each do |setting|
+            cli << "--#{setting}" << dir
           end
-
-          metadata =
-            if data[:metadata_file]
-              JSON.parse(File.read(data[:metadata_file]))
-            else
-              {}
+          Puppet.initialize_settings(cli)
+          Puppet::Pal.in_tmp_environment('bolt', modulepath: [BOLTLIB_PATH] + modulepath) do |pal|
+            pal.with_script_compiler do |compiler|
+              # Update once Pal includes task_signature (PUP-8202)
+              #task_type = compiler.task_signature(name)
+              task_type = Puppet.lookup(:loaders).private_environment_loader.load(:type, name)
+              task = compiler.create(task_type, args) if task_type
```

Comment:
PUP-8171 adds API for typechecking plans. A similar method for task (getting a signature and then checking the args against it is yet to be added).